File: backend/app/models/sla.py

from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.session import Base


class SLA(Base):
    __tablename__ = "slas"

    id: Mapped[int] = mapped_column(
        primary_key=True,
    )

    name: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
        unique=True,
    )

    description: Mapped[str | None] = mapped_column(
        String(500),
        nullable=True,
    )

    priority: Mapped[str] = mapped_column(
        String(30),
        nullable=False,
        index=True,
    )

    response_time_minutes: Mapped[int] = mapped_column(
        Integer,
        nullable=False,
    )

    resolution_time_minutes: Mapped[int] = mapped_column(
        Integer,
        nullable=False,
    )

    is_active: Mapped[bool] = mapped_column(
        Boolean,
        default=True,
        nullable=False,
        index=True,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )

    updated_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True),
        onupdate=func.now(),
        nullable=True,
    )

File: frontend/src/types/sla.ts
export interface SLA {
  id: number;
  name: string;
  description: string | null;
  priority: string;
  response_time_minutes: number;
  resolution_time_minutes: number;
  is_active: boolean;
  created_at: string;
  updated_at: string | null;
}

export interface SLACreateRequest {
  name: string;
  description?: string | null;
  priority: string;
  response_time_minutes: number;
  resolution_time_minutes: number;
  is_active?: boolean;
}

export interface SLAUpdateRequest {
  name?: string;
  description?: string | null;
  priority?: string;
  response_time_minutes?: number;
  resolution_time_minutes?: number;
  is_active?: boolean;
}
